src/m365/spo/commands/file/file-sharinglink-get.spec.ts

```diff
@@ -0,0 +1,195 @@
+    const actual = await command.validate({ options: { webUrl: webUrl, fileId: '12345', id: id } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if the fileId option is a valid GUID', async () => {
+    const actual = await command.validate({ options: { webUrl: webUrl, fileId: fileId, id: id } }, commandInfo);
+    assert(actual);
+  });
+
+  it('fails validation if the fileId or fileUrl option not specified', async () => {
+    const actual = await command.validate({ options: { webUrl: webUrl, id: id } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if both fileId and fileUrl options are specified', async () => {
+    const actual = await command.validate({ options: { webUrl: webUrl, fileId: fileId, fileUrl: fileUrl, id: id } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('gets a specific sharing link of a file by id', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `${webUrl}/_api/web/GetFileById('${fileId}')?$select=SiteId,VroomItemId,VroomDriveId`) {
+        return fileInformationResponse;
+      }
+
+      if (opts.url === `https://graph.microsoft.com/v1.0/sites/${fileInformationResponse.SiteId}/drives/${fileInformationResponse.VroomDriveID}/items/${fileInformationResponse.VroomItemID}/permissions/${id}`) {
+        return fileSharingLinkResponse;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        debug: true,
+        verbose: true,
+        webUrl: webUrl,
+        fileId: fileId,
+        id: id
+      }
+    });
+
+    assert(loggerLogSpy.calledWith(fileSharingLinkResponse));
+  });
+
+  it('gets a specific sharing link of a file by URL', async () => {
+    const fileServerRelativeUrl: string = urlUtil.getServerRelativePath(webUrl, fileUrl);
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `${webUrl}/_api/web/GetFileByServerRelativePath(decodedUrl='${formatting.encodeQueryParameter(fileServerRelativeUrl)}')?$select=SiteId,VroomItemId,VroomDriveId`) {
+        return fileInformationResponse;
+      }
+
+      if (opts.url === `https://graph.microsoft.com/v1.0/sites/${fileInformationResponse.SiteId}/drives/${fileInformationResponse.VroomDriveID}/items/${fileInformationResponse.VroomItemID}/permissions/${id}`) {
+        return fileSharingLinkResponse;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        debug: true,
+        verbose: true,
+        webUrl: webUrl,
+        fileUrl: fileUrl,
+        id: id
+      }
+    });
+
+    assert(loggerLogSpy.calledWith(fileSharingLinkResponse));
+  });
+
+  it('correctly handles random API error', async () => {
+    sinon.stub(request, 'get').callsFake(() => Promise.reject('An error has occurred'));
```

Comment:
Let's use a realistic error instead of a plain text one.

The API request will throw this kind of error:

`error: { 'odata.error': { message: { value: 'File Not Found.' } } } }`